src/communication.jl

```diff
@@ -208,14 +217,147 @@ function setup_distributed_memory_MPI(z_nelement_global,z_nelement_local,r_nelem
     return z_irank, z_nrank_per_group, z_comm, r_irank, r_nrank_per_group, r_comm
 end
 
+"""
+Function to take information from user about vpa vperp grids and 
+number of processes allocated to set up communicators for 
+precomputation of the Rosenbluth potential integration weights
+notation definitions:
+    - block: group of processes that share data with shared memory
+    - vpa group: group of processes that need to communicate data for vpa derivatives/integrals
+    - vperp group: group of processes that need to communicate data for vperp derivatives/integrals
+This routine assumes that the number of processes is selected by the user
+to match or be larger than the ratio 
+
+  nblocks = (vpa_nelement_global/vpa_nelement_local)*(vperp_nelement_global/vperp_nelement_local)
+  
+We also need to know (from user input) the maximum number of cores per shared memory region.
+A fraction of the cores will not contribute to the calculation, as we cannot guarantee that 
+the same number of cores is required for the rz parallelisation as the vpa vperp parallelisation 
+"""
+function setup_distributed_memory_MPI_for_weights_precomputation(vpa_nelement_global,vpa_nelement_local,
+               vperp_nelement_global,vperp_nelement_local, max_cores_per_block; printout=false)
```

Comment:
It is not used, but I thought it might be interesting, since it allows a distributed MPI setup for vpa and vperp where the number of cores does not match the element numbers exactly. I wouldn't remove it till we have decided how to finish the MPI optimisation for the collision operator.

Reply:
Are you happy with my resolution? https://github.com/mabarnes/moment_kinetics/pull/149/commits/a1e44155601a7c97aee4a517a7c2dc90c6b3aa22.

Reply:
Looks good to me. I haven't gone through the parallelisation logic in that function myself - happy to trust you.